framework/Shell/Actions/TPhpShellAction.php

```diff
@@ -16,34 +16,34 @@
 /**
  * Creates and run a Prado application in a PHP Shell.
  *
+ * @author Brad Anderson <belisoful[at]icloud[dot]com> shell refactor
  * @author Wei Zhuo <weizhuo[at]gmail[dot]com>
  * @package Prado\Shell\Actions
  * @since 3.0.5
  */
 class TPhpShellAction extends TShellAction
 {
 	protected $action = 'shell';
-	protected $parameters = [];
-	protected $optional = ['directory'];
-	protected $description = 'Runs a PHP interactive interpreter. Initializes the Prado application in the given [directory].';
-
+	protected $methods = ['index'];
+	protected $parameters = [null];
+	protected $optional = [null];
+	protected $description = [
+		'Provides PHP Interactive Shell Interpreter.',
+		'Runs a PHP interactive interpreter after Initializing the Prado application.'];
+	
 	/**
+	 * This runs the interactive PHP Shell
 	 * @param array $args parameters
 	 * @return bool
 	 */
-	public function performAction($args)
+	public function actionIndex($args)
 	{
-		if (count($args) > 1) {
-			$loaded = $this->initializePradoApplication($args[1]);
-			$this->getWriter()->flush();
-			if ($loaded === false) {
-				return true;
-			}
-		}
-
+		$this->getWriter()->flush();
+		
 		$shell = new \Psy\Shell();
 		$shell->setBoundObject($this);
```

Comment:
This used to bound $this to Prado::getApplication(), wether it now binds to the TPhpShellAction object.
Even if it's not related/caused by your PR, can you please restore it?

Reply:
I'll get on this.  Good eyes.

Reply:
The fix for no action parameter error (stated above) is a one line correction also.

Reply:
I didn't know how PsyShell worked or I would have bumped the bug earlier.

as an aside, the TPermissionManager is looking nice and firm.  I'm working on a shell action for permissions.